core/lib/types/src/fee_model.rs

```diff
@@ -236,9 +239,85 @@ pub struct FeeParamsV1 {
 
 #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub struct FeeParamsV2 {
-    pub config: FeeModelConfigV2,
-    pub l1_gas_price: u64,
-    pub l1_pubdata_price: u64,
+    config: FeeModelConfigV2,
+    l1_gas_price: u64,
+    l1_pubdata_price: u64,
+    conversion_ratio: BaseTokenConversionRatio,
+}
+
+impl FeeParamsV2 {
+    pub fn new(
+        config: FeeModelConfigV2,
+        l1_gas_price: u64,
+        l1_pubdata_price: u64,
+        conversion_ratio: BaseTokenConversionRatio,
+    ) -> Self {
+        Self {
+            config,
+            l1_gas_price,
+            l1_pubdata_price,
+            conversion_ratio,
+        }
+    }
+
+    /// Returns the fee model config with the minimal L2 gas price denominated in the chain's base token (WEI or equivalent).
+    pub fn config(&self) -> FeeModelConfigV2 {
+        FeeModelConfigV2 {
+            minimal_l2_gas_price: self.convert_to_base_token(self.config.minimal_l2_gas_price),
+            ..self.config
+        }
+    }
+
+    /// Returns the l1 gas price denominated in the chain's base token (WEI or equivalent).
+    pub fn l1_gas_price(&self) -> u64 {
+        self.convert_to_base_token(self.l1_gas_price)
+    }
+
+    /// Returns the l1 pubdata price denominated in the chain's base token (WEI or equivalent).
+    pub fn l1_pubdata_price(&self) -> u64 {
+        self.convert_to_base_token(self.l1_pubdata_price)
+    }
+
+    /// Converts the fee param to the base token.
+    fn convert_to_base_token(&self, price_in_wei: u64) -> u64 {
```

Comment:
We probably want metrics for stuff like base token price, price ratio, converted price for observability; without this information it may be hard to debug issues in production.
It doesn't have to be added in this PR, but please create a task and don't forget to add it (both in code and in grafana once we have at least one deployment).

Reply:
100%. There's already a task for all of this.